benchmark/test/test_framework.py.in

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env python3
+import subprocess
+import difflib
+import json
+import typing
+import re
+import pathlib
+import sys
+
+sourcepath = pathlib.Path("@CMAKE_CURRENT_SOURCE_DIR@")
+binpath = pathlib.Path("@PROJECT_BINARY_DIR@")
+generate = False
+if len(sys.argv) > 2 and sys.argv[2] == "--generate":
+    generate = True
+
+
+denumberify_paths = [
+    "time",
+    "bandwidth",
+    "flops",
+    "components",
+    "residual_norm",
+    "rhs_norm",
+    "max_relative_norm2",
+]
+empty_string_paths = ["error"]
+empty_array_paths = [
+    "recurrent_residuals",
+    "true_residuals",
+    "implicit_residuals",
+    "iteration_timestamps",
+]
+
+
+def sanitize_json_single(key, value, sanitize_all):
+    if key in empty_string_paths and isinstance(value, str):
+        return ""
+    if key in denumberify_paths and isinstance(value, float):
+        return 1.0
+    if key in denumberify_paths and isinstance(value, typing.Dict):
```

Comment:
Generally I think it would be bad if anything below here is actually reached  ...

Reply:
I don't follow. How would emptying residual norm arrays work otherwise?

Reply:
NVM.